rocAL/source/pipeline/master_graph.cpp

```diff
@@ -981,9 +981,10 @@ void MasterGraph::output_routine() {
             _ring_buffer.push();  // The data and metadata is now stored in output the ring_buffer, increases it's level by 1
         }
     } catch (const std::exception &e) {
-        ERR("Exception thrown in the process routine: " + STR(e.what()) + STR("\n"));
         _processing = false;
         _ring_buffer.release_all_blocked_calls();
+        release();
+        THROW("Exception thrown in the process routine: " + STR(e.what()) + STR("\n"));
```

Comment:
There shouldn't be a THOW here as this is the catch loop for all exceptions.

Reply:
Previously an ERROR statement was used but the ctest failed to capture those errors and resulted in tests passing, even though rocAL raised ERR.
Hence changed it to THROW so that ctest would capture it